src/client/microsoftAuth.js

```diff
@@ -34,6 +37,19 @@ async function authenticate (client, options) {
   }
   client.session = session
   client.username = profile.name
+  client.profileKeys = {}
+
+  if (certificates) {
+    client.profileKeys = {
+      publicPEM: certificates.keyPair.publicKey,
```

Comment:
Maybe some of this transformation could be handled in prismarine-auth

Reply:
Crypto should be done in its own class/util in nmp.

Reply:
Will move this to pauth as it should not be returning unabstracted 1:1 web request data